Update help docs for SPSE (#72)
MicrosoftDocs/OfficeDocs-SharePoint-PowerShell: sharepoint/sharepoint-server-ps/sharepoint-server/Set-SPWebApplication.md

 [-NotProvisionGlobally] [-AssignmentCollection <SPAssignmentCollection>] [-WhatIf] [-Confirm]
 [<CommonParameters>]
```

### UpdateIISSettings
```
Set-SPWebApplication [-Identity] <SPWebApplicationPipeBind> -Zone <SPUrlZone> [-SecureSocketsLayer]
 [-HostHeader <String>] [-Certificate <SPServerCertificatePipeBind>] [-UseServerNameIndication]
 [-AllowLegacyEncryption] -Port <Int32> [-Url <String>] [-NotProvisionGlobally]
 [-AssignmentCollection <SPAssignmentCollection>] [-WhatIf] [-Confirm] [<CommonParameters>]
```

### UpdateMailSettings
```
Set-SPWebApplication [-Identity] <SPWebApplicationPipeBind> -SMTPServer <String> [-SMTPServerPort <Int32>]
 [-OutgoingEmailAddress <String>] [-ReplyToEmailAddress <String>] [-SMTPCredentials <PSCredential>]
 [-DisableSMTPEncryption] [-Certificate <SPServerCertificatePipeBind>] [-NotProvisionGlobally]
 [-AssignmentCollection <SPAssignmentCollection>] [-WhatIf] [-Confirm] [<CommonParameters>]
```

## DESCRIPTION
This cmdlet contains more than one parameter set.
You may only use parameters from one parameter set, and you may not combine parameters from different parameter sets.

> For more information about how to use parameter sets, see [Cmdlet parameter sets](https://docs.microsoft.com/powershell/scripting/developer/cmdlet/cmdlet-parameter-sets).

Reviewer: ```suggestion
For more information about how to use parameter sets, see [Cmdlet parameter sets](/powershell/scripting/developer/cmdlet/cmdlet-parameter-sets).
```

## EXAMPLES

### EXAMPLE 1
```powershell
$credentials = Get-Credential
Set-SPWebApplication -Identity http://servername -SMTPServer mail.example.com -SMTPServerPort 587 -OutgoingEmailAddress user@example.com -ReplyToAddress replyto@example.com -SMTPCredentials $credentials
```

This example sets the http://servername web application to use the SMTP server mail.example.com, the SMTP server port 587, the from address user@example.com, the reply-to address replyto@example.com, and the credentials stored in the $credentials object.

### EXAMPLE 2
```powershell
Set-SPWebApplication -Identity http://servername -SMTPServer mail.example.com -OutgoingEmailAddress user@example.com -ReplyToAddress replyto@example.com -SMTPCredentials $null
```

This example sets the http://servername web application to use the SMTP server mail.example.com, the default SMTP server port, the from address user@example.com, the reply-to address replyto@example.com, and to connect to the SMTP server anonymously.

## PARAMETERS

### -AdditionalClaimProvider

> Adds a specific claim provider to the defined Web application.

Reviewer: ```suggestion
Adds a specific claim provider to the defined web application.
```

```yaml
Type: SPClaimProviderPipeBind[]
Parameter Sets: UpdateClaimSettings
Aliases:
Applicable: SharePoint Server Subscription Edition

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -AllowLegacyEncryption

> Specifies that older SSL and TLS protocol versions and cipher suites are allowed to be used with this IIS web site.

Reviewer: ```suggestion
Specifies that older SSL and TLS protocol versions and cipher suites are allowed to be used with this IIS website.
```

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -AssignmentCollection
Manages objects for the purpose of proper disposal.
Use of objects, such as SPWeb or SPSite, can use large amounts of memory and use of these objects in Windows PowerShell scripts requires proper memory management.
Using the SPAssignment object, you can assign objects to a variable and dispose of the objects after they are needed to free up memory.
When SPWeb, SPSite, or SPSiteAdministration objects are used, the objects are automatically disposed of if an assignment collection or the Global parameter is not used.

When the Global parameter is used, all objects are contained in the global store.
If objects are not immediately used, or disposed of by using the `Stop-SPAssignment` command, an out-of-memory scenario can occur.

```yaml
Type: SPAssignmentCollection
Parameter Sets: (All)
Aliases: 
Applicable: SharePoint Server Subscription Edition

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByValue)
Accept wildcard characters: False
```

### -AuthenticationMethod

> Use to set a Web application to classic Windows authentication.

Reviewer: ```suggestion
Use to set a web application to classic Windows authentication.
```

Accept wildcard characters: False
```

### -AuthenticationProvider

> Defines the authentication provider(s) that applies to the Web application.

Reviewer: ```suggestion
Defines the authentication provider(s) that applies to the web application.
```

Accept wildcard characters: False
```

### -Certificate

> Specifies the certificate that will be used for the Secure Sockets Layer (SSL) binding of this IIS web site, or for authenticating to an SMTP server to send email.
> 
> When configuring the Secure Sockets Layer (SSL) binding of this IIS web site, this parameter is only valid when used with the SecureSocketsLayer parameter.

Reviewer: ```suggestion
Specifies the certificate that will be used for the Secure Sockets Layer (SSL) binding of this IIS website, or for authenticating to an SMTP server to send email.

When configuring the Secure Sockets Layer (SSL) binding of this IIS website, this parameter is only valid when used with the SecureSocketsLayer parameter.
```

When configuring SMTP authentication, this parameter is only valid when the DisableSMTPEncryption parameter is not specified.

```yaml
Type: SPServerCertificatePipeBind
Parameter Sets: UpdateIISSettings, UpdateMailSettings
Aliases:
Applicable: SharePoint Server Subscription Edition

Required: False
Position: Named
Default value: None
Accept pipeline input: True (ByValue)
Accept wildcard characters: False
```

### -DefaultQuotaTemplate

> Specifies the new default site quota template for this Web application.

Reviewer: ```suggestion
Specifies the new default site quota template for this web application.
```

```yaml
Type: String
Parameter Sets: UpdateGeneralSettings
Aliases: 
Applicable: SharePoint Server Subscription Edition

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -DefaultTimeZone

> Specifies the default time zone for new site collections in this Web application.

Reviewer: ```suggestion
Specifies the default time zone for new site collections in this web application.
```

Accept wildcard characters: False
```

### -DisableSMTPEncryption
Specifies whether to turn on or off SMTP Encryption.

The default value is false.

```yaml
Type: SwitchParameter
Parameter Sets: UpdateMailSettings
Aliases: 
Applicable: SharePoint Server Subscription Edition

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -Force

> Suppresses confirmation messages involved in settings for a Web application.

Reviewer: ```suggestion
Suppresses confirmation messages involved in settings for a web application.
```

```yaml
Type: SwitchParameter
Parameter Sets: UpdateClaimSettings
Aliases:
Applicable: SharePoint Server Subscription Edition

Required: False
Position: Named
Default value: False
Accept pipeline input: False
Accept wildcard characters: False
```

### -HostHeader

> Specifies the host header binding for this IIS web site.

Reviewer: ```suggestion
Specifies the host header binding for this IIS website.
```

> A host header binding allows multiple IIS web sites to share the same port number.
> Web requests sent to a shared port number are routed to the correct IIS web site based on the value of the HTTP host header sent by the client.

Reviewer: ```suggestion
A host header binding allows multiple IIS websites to share the same port number.
Web requests sent to a shared port number are routed to the correct IIS website based on the value of the HTTP host header sent by the client.
```

> If no host header binding is specified, then all web requests sent to this port number will be routed to this IIS web site unless another IIS web site has a host header binding that matches the HTTP host header sent by the client.

Reviewer: ```suggestion
If no host header binding is specified, then all web requests sent to this port number will be routed to this IIS website unless another IIS website has a host header binding that matches the HTTP host header sent by the client.
```

Accept wildcard characters: False
```

### -Identity

> Specifies the name or URL of the Web application.

Reviewer: ```suggestion
Specifies the name or URL of the web application.
```

> The type must be a valid name, in the form WebApplication-1212, or URL, in the form http://server_name.

Reviewer 2: Rather than using http://server_name, could you use https://example.contoso.com? In my personal opinion, this encourages SSL + FQDN usage, guiding administrators towards best practice.

> The type must be a valid name, in the form WebApplication-1212, or URL, in the form http://server_name.

Reviewer: ```suggestion
The type must be a valid name, in the form WebApplication-1212, or URL, in the form https://example.contoso.com/.
```

```yaml
Type: SPWebApplicationPipeBind
Parameter Sets: (All)
Aliases: 
Applicable: SharePoint Server Subscription Edition

Required: True
Position: 1
Default value: None
Accept pipeline input: True (ByValue)
Accept wildcard characters: False
```

### -NotProvisionGlobally

> Skips the Web application provision process.

Reviewer 2: Correct me if I'm wrong, but doesn't this parameter simply provision the Web Application on the specific server the switch is run on? I.e., I would say something like "Only provisions the Web Application on the server the switch is specified on rather than all servers running the Foundation Web Service service instance".

Reviewer 3: @TroyStarr Could you help to take a look and give suggestions on this?

Reviewer 4: I agree with @tseward - we should clarify this to say something like "Only provisions the web application on the local server with the changes specified by this cmdlet. Web applications on other servers in the farm will not be provisioned with these changes."

> Skips the Web application provision process.

Reviewer: ```suggestion
Skips the web application provision process.
```

Accept wildcard characters: False
```

### -OutgoingEmailAddress

> Specifies the new outgoing e-mail address for e-mail messages sent from this Web application.

Reviewer: ```suggestion
Specifies the new outgoing e-mail address for e-mail messages sent from this web application.
```

Accept wildcard characters: False
```

### -Port

> Specifies the port on which this Web application can be accessed.

Reviewer: ```suggestion
Specifies the port on which this web application can be accessed.
```

Accept wildcard characters: False
```

### -SecureSocketsLayer

> Enables Secure Sockets Layer (SSL) encryption for this Web application.

Reviewer: ```suggestion
Enables Secure Sockets Layer (SSL) encryption for this web application.
```

> If you choose to use SSL, you must import a server certificate to SharePoint and assign it to the IIS web site for this web application.
> Until this is done, the Web application will be inaccessible from this IIS Web site.

Reviewer: ```suggestion
If you choose to use SSL, you must import a server certificate to SharePoint and assign it to the IIS website for this web application.
Until this is done, the web application will be inaccessible from this IIS website.
```

The default value is False.

> If this parameter is omitted or set to False, this Web application will use HTTP for the specified port.

Reviewer: ```suggestion
If this parameter is omitted or set to False, this web application will use HTTP for the specified port.
```

Accept wildcard characters: False
```

### -SMTPCredentials
Specifies credentials to authenticate to the SMTP server.
Set the value to $null to connect to the SMTP server anonymously.
If this parameter isn't specified, the existing authentication settings will be preserved.

> You must use the \`Set-SPApplicationCredentialKey\` PowerShell cmdlet to set an application credential key on each server in the farm before specifying credentials.

Reviewer 2: You may also want to mention that the App Cred Key needs to be the same value on each server in the farm.

Reviewer 3: @TroyStarr Could you please take a look at this and give suggestions on whether we need to add something related ro App Cred Key?

Reviewer 4: Sure, we can say something like "You must use the \`Set-SPApplicationCredentialKey\` PowerShell cmdlet to set an identical application credential key on each server in the farm before specifying credentials."

Accept wildcard characters: False
```

### -SMTPServer

> Specifies the new outbound SMTP server that this Web application will use.

Reviewer: ```suggestion
Specifies the new outbound SMTP server that this web application will use.
```

Set to $null to clear the server setting.

```yaml
Type: String
Parameter Sets: UpdateMailSettings
Aliases: 
Applicable: SharePoint Server Subscription Edition

Required: True
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -SMTPServerPort
Specifies a SMTP port value.

```yaml
Type: Int32
Parameter Sets: UpdateMailSettings
Aliases:

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -Url

> Specifies the load-balanced URL for the Web application zone.

Reviewer: ```suggestion
Specifies the load-balanced URL for the web application zone.
```

```yaml
Type: String
Parameter Sets: UpdateIISSettings
Aliases:
Applicable: SharePoint Server Subscription Edition

Required: False
Position: Named
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -UseServerNameIndication

> Specifies that the Secure Sockets Layer (SSL) binding of this IIS web site should use Server Name Indication (SNI).

Reviewer 2: This should be "Secure Socket Layer" rather than plural.

Reviewer 4: Actually, it should be plural "Secure Sockets Layer" rather than singular.  (I know, but that's what it's officially called...)

Reviewer 3: @tseward @TroyStarr So we would keep "Secure Sockets Layer" instead of changing.

> Specifies that the Secure Sockets Layer (SSL) binding of this IIS web site should use Server Name Indication (SNI).

Reviewer: ```suggestion
Specifies that the Secure Socket Layer (SSL) binding of this IIS website should use Server Name Indication (SNI).
```

Server Name Indication allows multiple IIS web sites with unique host headers and unique server certificates to share the same SSL port.

> If Server Name Indication isn't used, all IIS web sites sharing the same SSL port must share the same server certificate.

Reviewer 2: Rather than 'server certificate', I would specify "SSL certificate'.

> If Server Name Indication isn't used, all IIS web sites sharing the same SSL port must share the same server certificate.

Reviewer: ```suggestion
If Server Name Indication isn't used, all IIS websites sharing the same SSL port must share the same SSL certificate.
```
